src/Datadog.Trace/Agent/MessagePack/TraceRequestMessagePackFormatter.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using Datadog.Trace.Agent.NamedPipes;
+using Datadog.Trace.Vendors.MessagePack;
+using Datadog.Trace.Vendors.MessagePack.Formatters;
+
+namespace Datadog.Trace.Agent.MessagePack
+{
+    internal class TraceRequestMessagePackFormatter : IMessagePackFormatter<TraceRequest>
+    {
+        public int Serialize(ref byte[] bytes, int offset, TraceRequest value, IFormatterResolver formatterResolver)
+        {
+            offset += value.WriteHeaders(ref bytes, offset);
+            offset += value.WriteTraces(ref bytes, offset);
```

Comment:
Similar comment to the span formatter: `WriteHeaders()` and `WriteTraces()` only access public properties, so those could be here in the formatter instead of "polluting" the `TraceRequest` poco.